assets/src/stories-editor/components/block-mover/block-draggable.js

```diff
@@ -10,28 +10,108 @@ import PropTypes from 'prop-types';
 /**
  * WordPress dependencies
  */
-import { withSelect } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
+import { useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Draggable from './draggable';
+import { isBlockAllowedOnPage, useMoveBlockToPage } from '../../helpers';
+
+const BlockDraggable = ( { children, clientId, blockName, blockElementId, onDragStart, onDragEnd } ) => {
+	const { rootClientId } = useSelect( ( select ) => select( 'core/block-editor' ).getBlockRootClientId( clientId ) );
+	const { index } = useSelect( ( select ) => select( 'core/block-editor' ).getBlockIndex( clientId, rootClientId ) );
 
-const BlockDraggable = ( { children, clientId, blockName, rootClientId, blockElementId, index, onDragStart, onDragEnd } ) => {
 	const transferData = {
 		type: 'block',
 		srcIndex: index,
 		srcRootClientId: rootClientId,
 		srcClientId: clientId,
 	};
 
+	const { moveBlockToPage, getPageByOffset } = useMoveBlockToPage( clientId );
+
+	// This holds the currently highlighted element, if any
+	const hoverElement = useRef( { pageId: null, element: null, classes: [] } );
+
+	/**
+	 * Clear highlighting of pages.
+	 */
+	const clearHighlight = () => {
+		// Unhighlight old highlighted page.
+		if ( hoverElement.current.element ) {
+			hoverElement.current.element.classList.remove( ...hoverElement.current.classes );
+		}
+
+		// Reset current hover element
+		hoverElement.current = { pageId: null, element: null, classes: [] };
```

Comment:
This clears out all classes. Are they not classes on the element we might want to keep?

Reply:
This doesn't actually manipulate any element. The `classes` property here only remembers which classes were set on the last `element` in use in order to remove them again.